src/constants.ts

```diff
@@ -23,9 +23,9 @@ export const mergeableImportFlavors = [
     importFlavorType,
 ] as const;
 
-export const BUILTIN_MODULES_REGEX_STR = `^(?:node:)?(?:${builtinModules.join(
+export const BUILTIN_MODULES_REGEX_STR = `(^(?:node:)?(?:${builtinModules.join(
     '|',
-)})$`;
+)})$)|(^node:test$)`; // NOTE: 'test' is not included in `builtinModules` so explicitly check for "node:test"
```

Comment:
My only small suggestion is to add a link to the issue: https://github.com/nodejs/node/issues/42785.  Maybe someday it will be solved, and we will be able to remove this workaround.